spec/helpers/ajax-helper.ts

```diff
@@ -218,4 +226,33 @@ export class MockXMLHttpRequest {
       }
     });
   }
-}
\ No newline at end of file
+}
+
+export class MockXMLHttpRequestInternetExplorer extends MockXMLHttpRequest {
+  constructor() {
+    super();
+  }
+
+  private mockHttp204() {
+    this.responseType = '';
+    this.responseText = '';
+    this.response = '';
+  }
+
+  protected jsonResponseValue(response: any) {
+    if (this.status == 204) {
+      this.mockHttp204();
+      return;
+    }
+    return super.jsonResponseValue(response);
+  }
+
+  protected defaultResponseValue() {
```

Comment:
Well, if it is used incorrectly, (test respondWith ```'responseType': 'json',```) this will give a false positive. Basically, my intention was to keep all IE mock behavior inside of the class. Otherwise every test writer has to know how IE behaves.